[2022-05-30 09:48:49,653] {taskinstance.py:903} INFO - Dependencies all met for <TaskInstance: my_spark_dag.process_songs 2022-05-30T09:48:47.858263+00:00 [queued]>
[2022-05-30 09:48:49,661] {taskinstance.py:903} INFO - Dependencies all met for <TaskInstance: my_spark_dag.process_songs 2022-05-30T09:48:47.858263+00:00 [queued]>
[2022-05-30 09:48:49,661] {taskinstance.py:1095} INFO - 
--------------------------------------------------------------------------------
[2022-05-30 09:48:49,661] {taskinstance.py:1096} INFO - Starting attempt 1 of 4
[2022-05-30 09:48:49,661] {taskinstance.py:1097} INFO - 
--------------------------------------------------------------------------------
[2022-05-30 09:48:49,667] {taskinstance.py:1115} INFO - Executing <Task(PythonOperator): process_songs> on 2022-05-30T09:48:47.858263+00:00
[2022-05-30 09:48:49,669] {standard_task_runner.py:52} INFO - Started process 10684 to run task
[2022-05-30 09:48:49,671] {standard_task_runner.py:76} INFO - Running: ['***', 'tasks', 'run', 'my_spark_dag', 'process_songs', '2022-05-30T09:48:47.858263+00:00', '--job-id', '5', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/spark_dag.py', '--cfg-path', '/tmp/tmpngw3bdzu', '--error-file', '/tmp/tmp9nttmjhz']
[2022-05-30 09:48:49,673] {standard_task_runner.py:77} INFO - Job 5: Subtask process_songs
[2022-05-30 09:48:49,698] {logging_mixin.py:109} INFO - Running <TaskInstance: my_spark_dag.process_songs 2022-05-30T09:48:47.858263+00:00 [running]> on host 1367f8687ad2
[2022-05-30 09:48:49,726] {taskinstance.py:1254} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=kirill
AIRFLOW_CTX_DAG_ID=my_spark_dag
AIRFLOW_CTX_TASK_ID=process_songs
AIRFLOW_CTX_EXECUTION_DATE=2022-05-30T09:48:47.858263+00:00
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-05-30T09:48:47.858263+00:00
[2022-05-30 09:49:57,523] {taskinstance.py:1463} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1165, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1283, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1313, in _execute_task
    result = task_copy.execute(context=context)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/operators/python.py", line 150, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/operators/python.py", line 161, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/spark_funcs.py", line 31, in process_song_data
    spark = create_spark_session('172.20.0.1:7077')
  File "/opt/airflow/dags/spark_funcs.py", line 13, in create_spark_session
    .master("spark://{}".format(master_address))\
  File "/home/airflow/.local/lib/python3.6/site-packages/pyspark/sql/session.py", line 228, in getOrCreate
    sc = SparkContext.getOrCreate(sparkConf)
  File "/home/airflow/.local/lib/python3.6/site-packages/pyspark/context.py", line 392, in getOrCreate
    SparkContext(conf=conf or SparkConf())
  File "/home/airflow/.local/lib/python3.6/site-packages/pyspark/context.py", line 147, in __init__
    conf, jsc, profiler_cls)
  File "/home/airflow/.local/lib/python3.6/site-packages/pyspark/context.py", line 209, in _do_init
    self._jsc = jsc or self._initialize_context(self._conf._jconf)
  File "/home/airflow/.local/lib/python3.6/site-packages/pyspark/context.py", line 329, in _initialize_context
    return self._jvm.JavaSparkContext(jconf)
  File "/home/airflow/.local/lib/python3.6/site-packages/py4j/java_gateway.py", line 1586, in __call__
    answer, self._gateway_client, None, self._fqn)
  File "/home/airflow/.local/lib/python3.6/site-packages/py4j/protocol.py", line 328, in get_return_value
    format(target_id, ".", name), value)
py4j.protocol.Py4JJavaError: An error occurred while calling None.org.apache.spark.api.java.JavaSparkContext.
: java.lang.IllegalArgumentException: requirement failed: Can only call getServletHandlers on a running MetricsSystem
	at scala.Predef$.require(Predef.scala:281)
	at org.apache.spark.metrics.MetricsSystem.getServletHandlers(MetricsSystem.scala:89)
	at org.apache.spark.SparkContext.<init>(SparkContext.scala:603)
	at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:238)
	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
	at java.base/java.lang.Thread.run(Thread.java:829)

[2022-05-30 09:49:57,526] {taskinstance.py:1513} INFO - Marking task as UP_FOR_RETRY. dag_id=my_spark_dag, task_id=process_songs, execution_date=20220530T094847, start_date=20220530T094849, end_date=20220530T094957
[2022-05-30 09:49:57,561] {local_task_job.py:151} INFO - Task exited with return code 1
[2022-05-30 09:49:57,577] {local_task_job.py:261} INFO - 0 downstream tasks scheduled from follow-on schedule check
